File: Codeforces/R171/B_Books.java
package R171;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

/**
 * Created on 2024-09-09 01:34:18
 * 
 * @author macinchang
 */
public class B_Books {

    public static void main(String[] args) throws Exception {
        Scanner scanner;
        if (System.getProperty("ONLINE_JUDGE") == null) {
            scanner = new Scanner("in.txt");
        } else {
            scanner = new Scanner(System.in);
        }
        Solver solver = new Solver(scanner);
        solver.solve();

    }

    public static class Solver {
        private Scanner sc;

        Solver(Scanner sc) {
            this.sc = sc;
        }

        public void solve() throws IOException {
            int n = sc.nextInt(), t = sc.nextInt();
            int cur = 0;
            int ans = 0;
            int left = 0;
            int[] books = new int[n];
            for (int i = 0; i < n; i++) {
                books[i] = sc.nextInt();
                if (books[i] + cur <= t) {
                    cur += books[i];
                    continue;
                }
                ans = Math.max(ans, i - left);
                cur += books[i];
                while (cur > t) {
                    cur -= books[left];
                    ++left;
                    if (left > i) {
                        break;
                    }
                }
            }
            if(cur <= t) {
                ans = Math.max(ans, n - left);
            }
            System.out.println(ans);
        }
    }

    static class Scanner {
        StringTokenizer st;
        BufferedReader br;

        public Scanner(String string) throws FileNotFoundException {
            br = new BufferedReader(new FileReader(string));
        }

        public Scanner(FileReader r) {
            br = new BufferedReader(r);
        }

        public Scanner(InputStream in) {
            br = new BufferedReader(new InputStreamReader(in));
        }

        public String next() throws IOException {
            while (st == null || !st.hasMoreTokens())
                st = new StringTokenizer(br.readLine());
            return st.nextToken();
        }

        public int nextInt() throws IOException {
            return Integer.parseInt(next());
        }

        public long nextLong() throws IOException {
            return Long.parseLong(next());
        }

        public String nextLine() throws IOException {
            return br.readLine();
        }

        public double nextDouble() throws IOException {
            return Double.parseDouble(next());
        }

        public boolean ready() throws IOException {
            return br.ready();
        }
    }
}

File: Codeforces/R171/B_Books.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	var scanner *Scanner
	if os.Getenv("ONLINE_JUDGE") == "" {
		file, err := os.Open("in.txt")
		if err != nil {
			panic(err)
		}
		defer file.Close()
		scanner = NewScanner(file)
	} else {
		scanner = NewScanner(os.Stdin)
	}

	solver := NewSolver(scanner)
	solver.solve()
}

type Solver struct {
	sc *Scanner
}

func NewSolver(sc *Scanner) *Solver {
	return &Solver{sc: sc}
}

func (s *Solver) solve() {
	n := s.sc.nextInt()
	t := s.sc.nextInt()
	cur := 0
	ans := 0
	left := 0
	books := make([]int, n)

	for i := 0; i < n; i++ {
		books[i] = s.sc.nextInt()
		if books[i]+cur <= t {
			cur += books[i]
			continue
		}
		if i-left > ans {
			ans = i - left
		}
		cur += books[i]
		for cur > t {
			cur -= books[left]
			left++
			if left > i {
				break
			}
		}
	}
	if cur <= t && n-left > ans {
		ans = n - left
	}
	fmt.Println(ans)
}

type Scanner struct {
	scanner *bufio.Scanner
}

func NewScanner(r *os.File) *Scanner {
	return &Scanner{scanner: bufio.NewScanner(r)}
}

func (s *Scanner) next() string {
	s.scanner.Scan()
	return s.scanner.Text()
}

func (s *Scanner) nextInt() int {
	val, _ := strconv.Atoi(s.next())
	return val
}

func (s *Scanner) nextLine() string {
	s.scanner.Scan()
	return s.scanner.Text()
}
